test_xml_insert failure

```
============================= test session starts ==============================
platform darwin -- Python 3.8.5, pytest-6.1.1, py-1.9.0, pluggy-0.13.1 -- /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target/env/bin/python
cachedir: .pytest_cache
rootdir: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target, configfile: setup.cfg, testpaths: src/test/python
plugins: forked-1.3.0, xdist-2.1.0, mockito-0.0.4
gw0 I / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I / gw8 I / gw9 I / gw10 I / gw11 I
[gw0] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw1] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw2] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw3] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw4] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw5] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw6] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw7] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw8] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw9] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw10] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw11] darwin Python 3.8.5 cwd: /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target
[gw3] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw7] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw8] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw9] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw10] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw11] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw2] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw0] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw4] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw1] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw5] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
[gw6] Python 3.8.5 (default, Jul 21 2020, 10:48:26)  -- [Clang 11.0.3 (clang-1103.0.32.62)]
gw0 [78] / gw1 [78] / gw2 [78] / gw3 [78] / gw4 [78] / gw5 [78] / gw6 [78] / gw7 [78] / gw8 [78] / gw9 [78] / gw10 [78] / gw11 [78]

scheduling tests via LoadScheduling

src/test/python/test_binaries.py::test_opengrok_binary[opengrok-java]
src/test/python/test_binaries.py::test_opengrok_binary[opengrok-indexer]
src/test/python/test_binaries.py::test_opengrok_binary[opengrok-groups]
src/test/python/test_binaries.py::test_opengrok_binary[opengrok-deploy]
src/test/python/test_binaries.py::test_opengrok_binary[opengrok-reindex-project]
src/test/python/test_command.py::test_subst_append_exclsubst
[gw6] [  1%] PASSED src/test/python/test_command.py::test_subst_append_exclsubst
src/test/python/test_binaries.py::test_opengrok_binary[opengrok-config-merge]
src/test/python/test_binaries.py::test_opengrok_binary[opengrok-projadm]
src/test/python/test_binaries.py::test_opengrok_binary[opengrok]
src/test/python/test_binaries.py::test_opengrok_binary[opengrok-sync]
src/test/python/test_command.py::test_subst_append_default
[gw5] [  2%] PASSED src/test/python/test_command.py::test_subst_append_default
src/test/python/test_binaries.py::test_opengrok_binary[opengrok-mirror]
src/test/python/test_command.py::test_long_output
src/test/python/test_command.py::test_stderr
[gw5] [  3%] PASSED src/test/python/test_command.py::test_stderr
src/test/python/test_command_sequence.py::test_str
[gw5] [  5%] PASSED src/test/python/test_command_sequence.py::test_str
src/test/python/test_command_sequence.py::test_run_retcodes
[gw5] [  6%] PASSED src/test/python/test_command_sequence.py::test_run_retcodes
src/test/python/test_command_sequence.py::test_terminate_after_non_zero_code
[gw5] [  7%] PASSED src/test/python/test_command_sequence.py::test_terminate_after_non_zero_code
src/test/python/test_command_sequence.py::test_exit_2_handling
[gw5] [  8%] PASSED src/test/python/test_command_sequence.py::test_exit_2_handling
src/test/python/test_command_sequence.py::test_driveon_flag
[gw6] [ 10%] PASSED src/test/python/test_command.py::test_long_output
src/test/python/test_command.py::test_resource_limits
[gw5] [ 11%] PASSED src/test/python/test_command_sequence.py::test_driveon_flag
src/test/python/test_command_sequence.py::test_project_subst
[gw5] [ 12%] PASSED src/test/python/test_command_sequence.py::test_project_subst
src/test/python/test_command_sequence.py::test_cleanup
[gw6] [ 14%] PASSED src/test/python/test_command.py::test_resource_limits
src/test/python/test_command_sequence.py::test_cleanup_exception
[gw6] [ 15%] PASSED src/test/python/test_command_sequence.py::test_cleanup_exception
src/test/python/test_indexer.py::test_merge_properties
[gw6] [ 16%] PASSED src/test/python/test_indexer.py::test_merge_properties
src/test/python/test_mirror.py::test_empty_project_configuration
[gw6] [ 17%] PASSED src/test/python/test_mirror.py::test_empty_project_configuration
src/test/python/test_mirror.py::test_none_project_configuration
[gw6] [ 19%] PASSED src/test/python/test_mirror.py::test_none_project_configuration
src/test/python/test_mirror.py::test_invalid_project_config_proxy
[gw6] [ 20%] PASSED src/test/python/test_mirror.py::test_invalid_project_config_proxy
src/test/python/test_mirror.py::test_invalid_project_config_option
[gw6] [ 21%] PASSED src/test/python/test_mirror.py::test_invalid_project_config_option
src/test/python/test_mirror.py::test_invalid_project_config_regex
[gw6] [ 23%] PASSED src/test/python/test_mirror.py::test_invalid_project_config_regex
src/test/python/test_mirror.py::test_invalid_project_config_hookdir
[gw6] [ 24%] PASSED src/test/python/test_mirror.py::test_invalid_project_config_hookdir
src/test/python/test_mirror.py::test_invalid_project_config_ignoredrepos
[gw6] [ 25%] PASSED src/test/python/test_mirror.py::test_invalid_project_config_ignoredrepos
src/test/python/test_mirror.py::test_invalid_project_config_hooks
[gw6] [ 26%] PASSED src/test/python/test_mirror.py::test_invalid_project_config_hooks
[gw5] [ 28%] PASSED src/test/python/test_command_sequence.py::test_cleanup
src/test/python/test_mirror.py::test_invalid_project_config_hooknames
src/test/python/test_command_sequence.py::test_restful_fail
[gw6] [ 29%] PASSED src/test/python/test_mirror.py::test_invalid_project_config_hooknames
[gw5] [ 30%] PASSED src/test/python/test_command_sequence.py::test_restful_fail
src/test/python/test_mirror.py::test_invalid_project_config_nonexec_hook
src/test/python/test_mirror.py::test_valid_project_config_hook
[gw6] [ 32%] PASSED src/test/python/test_mirror.py::test_invalid_project_config_nonexec_hook
src/test/python/test_mirror.py::test_invalid_config_option
[gw5] [ 33%] PASSED src/test/python/test_mirror.py::test_valid_project_config_hook
[gw6] [ 34%] PASSED src/test/python/test_mirror.py::test_invalid_config_option
src/test/python/test_mirror.py::test_incoming_retval
src/test/python/test_mirror.py::test_valid_config
[gw5] [ 35%] PASSED src/test/python/test_mirror.py::test_valid_config
src/test/python/test_mirror.py::test_disabled_command_api
[gw5] [ 37%] PASSED src/test/python/test_mirror.py::test_disabled_command_api
src/test/python/test_mirror.py::test_disabled_command_api_text_append
[gw5] [ 38%] PASSED src/test/python/test_mirror.py::test_disabled_command_api_text_append
src/test/python/test_mirror.py::test_disabled_command_run
[gw5] [ 39%] PASSED src/test/python/test_mirror.py::test_disabled_command_run
src/test/python/test_mirror.py::test_disabled_command_run_args
[gw5] [ 41%] PASSED src/test/python/test_mirror.py::test_disabled_command_run_args
src/test/python/test_mirror.py::test_mirror_project_timeout
[gw5] [ 42%] PASSED src/test/python/test_mirror.py::test_mirror_project_timeout
src/test/python/test_mirror.py::test_get_repos_for_project
[gw5] [ 43%] PASSED src/test/python/test_mirror.py::test_get_repos_for_project
src/test/python/test_parsers.py::test_str2bool_exception
[gw5] [ 44%] PASSED src/test/python/test_parsers.py::test_str2bool_exception
src/test/python/test_parsers.py::test_str2bool_exception_empty_str
[gw5] [ 46%] PASSED src/test/python/test_parsers.py::test_str2bool_exception_empty_str
src/test/python/test_parsers.py::test_str2bool_bool
[gw5] [ 47%] PASSED src/test/python/test_parsers.py::test_str2bool_bool
src/test/python/test_parsers.py::test_str2bool
[gw5] [ 48%] PASSED src/test/python/test_parsers.py::test_str2bool
src/test/python/test_repofactory.py::test_repofactory_timeout
[gw9] [ 50%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-config-merge]
src/test/python/test_command.py::test_work_dir
[gw9] [ 51%] PASSED src/test/python/test_command.py::test_work_dir
src/test/python/test_restful.py::test_unknown_verb
[gw9] [ 52%] PASSED src/test/python/test_restful.py::test_unknown_verb
src/test/python/test_restful.py::test_headers
[gw9] [ 53%] PASSED src/test/python/test_restful.py::test_headers
src/test/python/test_restful.py::test_restful_fail
[gw9] [ 55%] PASSED src/test/python/test_restful.py::test_restful_fail
src/test/python/test_restful.py::test_invalid_command_negative
[gw9] [ 56%] PASSED src/test/python/test_restful.py::test_invalid_command_negative
src/test/python/test_versions.py::test_opengrok_version[opengrok]
[gw3] [ 57%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok]
src/test/python/test_command.py::test_subst_append_exclsubst_nosubst
[gw3] [ 58%] PASSED src/test/python/test_command.py::test_subst_append_exclsubst_nosubst
src/test/python/test_versions.py::test_opengrok_version[opengrok-groups]
[gw2] [ 60%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-mirror]
src/test/python/test_command.py::test_retcode[/usr/bin/true-/usr/bin/false]
[gw2] [ 61%] PASSED src/test/python/test_command.py::test_retcode[/usr/bin/true-/usr/bin/false]
src/test/python/test_versions.py::test_opengrok_version[opengrok-deploy]
[gw9] [ 62%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok]
src/test/python/test_versions.py::test_opengrok_version[opengrok-indexer]
[gw0] [ 64%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-projadm]
src/test/python/test_command.py::test_command_to_str
[gw0] [ 65%] PASSED src/test/python/test_command.py::test_command_to_str
src/test/python/test_versions.py::test_opengrok_version[opengrok-projadm]
[gw11] [ 66%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-java]
src/test/python/test_command.py::test_retcode[/bin/true-/bin/false]
[gw11] [ 67%] SKIPPED src/test/python/test_command.py::test_retcode[/bin/true-/bin/false]
src/test/python/test_versions.py::test_opengrok_version[opengrok-sync]
[gw3] [ 69%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-groups]
src/test/python/test_versions.py::test_opengrok_version[opengrok-config-merge]
[gw8] [ 70%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-groups]
src/test/python/test_command.py::test_getoutput
[gw8] [ 71%] PASSED src/test/python/test_command.py::test_getoutput
[gw10] [ 73%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-deploy]
src/test/python/test_command.py::test_env
[gw10] [ 74%] PASSED src/test/python/test_command.py::test_env
[gw6] [ 75%] FAILED src/test/python/test_mirror.py::test_incoming_retval
src/test/python/test_mirror.py::test_empty_project_config
[gw2] [ 76%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-deploy]
[gw6] [ 78%] PASSED src/test/python/test_mirror.py::test_empty_project_config
src/test/python/test_versions.py::test_opengrok_version[opengrok-java]
[gw7] [ 79%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-indexer]
src/test/python/test_command.py::test_execute_nonexistent
[gw7] [ 80%] PASSED src/test/python/test_command.py::test_execute_nonexistent
[gw9] [ 82%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-indexer]
src/test/python/test_versions.py::test_opengrok_version[opengrok-mirror]
[gw0] [ 83%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-projadm]
src/test/python/test_versions.py::test_opengrok_version[opengrok-reindex-project]
[gw4] [ 84%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-reindex-project]
src/test/python/test_command.py::test_command_timeout
[gw3] [ 85%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-config-merge]
src/test/python/test_xml.py::test_invalid_xml
[gw3] [ 87%] PASSED src/test/python/test_xml.py::test_invalid_xml
[gw11] [ 88%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-sync]
src/test/python/test_xml.py::test_xml_insert
[gw1] [ 89%] PASSED src/test/python/test_binaries.py::test_opengrok_binary[opengrok-sync]
src/test/python/test_command.py::test_command_notimeout
[gw11] [ 91%] FAILED src/test/python/test_xml.py::test_xml_insert
[gw2] [ 92%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-java]
[gw9] [ 93%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-mirror]
[gw0] [ 94%] PASSED src/test/python/test_versions.py::test_opengrok_version[opengrok-reindex-project]
[gw1] [ 96%] PASSED src/test/python/test_command.py::test_command_notimeout
[gw5] [ 97%] PASSED src/test/python/test_repofactory.py::test_repofactory_timeout
src/test/python/test_restful.py::test_replacement
[gw5] [ 98%] PASSED src/test/python/test_restful.py::test_replacement
[gw4] [100%] PASSED src/test/python/test_command.py::test_command_timeout

=================================== FAILURES ===================================
_____________________________ test_incoming_retval _____________________________
[gw6] darwin -- Python 3.8.5 /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target/env/bin/python
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target/env/lib/python3.8/site-packages/opengrok_tools-1.4.8-py3.8.egg/opengrok_tools/mirror.py", line 74, in worker
    return mirror_project(config, project_name,
  File "/Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target/env/lib/python3.8/site-packages/opengrok_tools-1.4.8-py3.8.egg/opengrok_tools/utils/mirror.py", line 392, in mirror_project
    repos = get_repos_for_project(project_name,
  File "/Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target/env/lib/python3.8/site-packages/opengrok_tools-1.4.8-py3.8.egg/opengrok_tools/utils/mirror.py", line 82, in get_repos_for_project
    for repo_path in get_repos(logger, project_name, uri):
TypeError: 'NoneType' object is not iterable
"""

The above exception was the direct cause of the following exception:

monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x10f74d700>

    @pytest.mark.skipif(not os.name.startswith("posix"), reason="requires posix")
    def test_incoming_retval(monkeypatch):
        """
        Test that the special CONTINUE_EXITVAL value bubbles all the way up to
        the mirror.py return value.
        """

        class MockResponse:

            # mock json() method always returns a specific testing dictionary
            @staticmethod
            def json():
                return "true"

            @staticmethod
            def raise_for_status():
                pass

        with tempfile.TemporaryDirectory() as source_root:
            repo_name = "parent_repo"
            repo_path = os.path.join(source_root, repo_name)
            cloned_repo_name = "cloned_repo"
            cloned_repo_path = os.path.join(source_root, cloned_repo_name)
            project_name = "foo"    # does not matter for this test

            os.mkdir(repo_path)

            def mock_get_repos(*args, **kwargs):
                return [get_repository(cloned_repo_path,
                                       "git", project_name)]

            def mock_get(*args, **kwargs):
                return MockResponse()

            # Clone a Git repository so that it can pull.
            repo = Repo.init(repo_path)
            new_file_path = os.path.join(repo_path, 'foo')
            with open(new_file_path, 'w'):
                pass
            assert os.path.isfile(new_file_path)
            index = repo.index
            index.add([new_file_path])
            index.commit("add file")
            repo.clone(cloned_repo_path)

            with monkeypatch.context() as m:
                m.setattr(sys, 'argv', ['prog', "-I", project_name])

                # With mocking done via pytest it is necessary to patch
                # the call-site rather than the absolute object path.
                m.setattr("opengrok_tools.mirror.get_config_value",
                          lambda x, y, z: source_root)
                m.setattr("opengrok_tools.utils.mirror.get_repos_for_project",
                          mock_get_repos)
                m.setattr("opengrok_tools.utils.mirror.do_api_call", mock_get)

>               assert opengrok_tools.mirror.main() == CONTINUE_EXITVAL

src/test/python/test_mirror.py:183:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target/env/lib/python3.8/site-packages/opengrok_tools-1.4.8-py3.8.egg/opengrok_tools/mirror.py:185: in main
    project_results = pool.map(worker, worker_args, 1)
/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py:364: in map
    return self._map_async(func, iterable, mapstar, chunksize).get()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <multiprocessing.pool.MapResult object at 0x10f7221c0>, timeout = None

    def get(self, timeout=None):
        self.wait(timeout)
        if not self.ready():
            raise TimeoutError
        if self._success:
            return self._value
        else:
>           raise self._value
E           TypeError: 'NoneType' object is not iterable

/usr/local/Cellar/python@3.8/3.8.5/Frameworks/Python.framework/Versions/3.8/lib/python3.8/multiprocessing/pool.py:771: TypeError
----------------------------- Captured stderr call -----------------------------
could not get repositories for foo
_______________________________ test_xml_insert ________________________________
[gw11] darwin -- Python 3.8.5 /Users/ahornace/IdeaProjects/opengrok/opengrok-tools/target/env/bin/python

    @pytest.mark.skipif(platform.system() == 'Windows',
                        reason="broken on Windows")
    def test_xml_insert():
        with open(os.path.join(DIR_PATH, "web.xml")) as base_xml:
            out = insert_file(base_xml.read(),
                              os.path.join(DIR_PATH, "insert.xml"))
            with open(os.path.join(DIR_PATH, "new.xml")) as expected_xml:
                # TODO: this should really be comparing XML trees
>               assert out == expected_xml.read()
E               assert '<?xml versio...n\n</web-app>' == '<?xml versio...n\n</web-app>'
E                   <?xml version='1.0' encoding='utf8'?>
E                 - <web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="3.1" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee          http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd">
E                 ?                                                                                                           --------------
E                 + <web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee          http:...
E
E                 ...Full output truncated (45 lines hidden), use '-vv' to show

src/test/python/test_xml.py:43: AssertionError
=========================== short test summary info ============================
FAILED src/test/python/test_mirror.py::test_incoming_retval - TypeError: 'Non...
FAILED src/test/python/test_xml.py::test_xml_insert - assert '<?xml versio......
=================== 2 failed, 75 passed, 1 skipped in 13.32s ===================
[ERROR] Command execution failed.
org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1)
    at org.apache.commons.exec.DefaultExecutor.executeInternal (DefaultExecutor.java:404)
    at org.apache.commons.exec.DefaultExecutor.execute (DefaultExecutor.java:166)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:804)
    at org.codehaus.mojo.exec.ExecMojo.executeCommandLine (ExecMojo.java:751)
    at org.codehaus.mojo.exec.ExecMojo.execute (ExecMojo.java:313)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:956)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:288)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:192)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:564)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:50)
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for OpenGrok 1.4.8:
[INFO]
[INFO] OpenGrok ........................................... SUCCESS [  4.114 s]
[INFO] OpenGrok Indexer ................................... SUCCESS [07:10 min]
[INFO] OpenGrok authorization plugins ..................... SUCCESS [ 53.703 s]
[INFO] OpenGrok Suggester ................................. SUCCESS [ 24.909 s]
[INFO] OpenGrok Web ....................................... SUCCESS [03:03 min]
[INFO] OpenGrok tools ..................................... FAILURE [01:02 min]
[INFO] Distribution ....................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12:38 min
[INFO] Finished at: 2020-10-19T15:10:01+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.6.0:exec (Test python package) on project opengrok-tools: Command execution failed.: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :opengrok-tools
```

--- a/opengrok-tools/src/test/python/web.xml
+++ b/opengrok-tools/src/test/python/web.xml
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         version="3.1"
          xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
-         http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
-         version="3.1">
+         http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd">
 
     <display-name>OpenGrok</display-name>
     <description>A wicked fast source browser</description>

